### PR TITLE
Fms2_io: Reproduce the null character mpp_io bug

### LINKED ISSUE #578: mpp_io bug: Not reading lat_bnds correctly which leads to data_override answer changes
**Describe the bug**
If, _for some reason_, you are using a file that has a c null character in the string attributes, the *bnds are not going to be read correctly when you are using mpp io which will lead to "wrong" answers in the corresponding data_override calls for that file ...  

The *_bnds are all read and buffered when `mpp_open` is called, which calls `mpp_read_meta`  

In `mpp_read_meta` there is this logic:
https://github.com/NOAA-GFDL/FMS/blob/2ff5e09985ffa7a5a8a51c40c4ffc5619128bf8e/mpp/include/mpp_io_read.inc#L813
which when it is true, it reads the *_bnds, otherwise it will end up here:
https://github.com/NOAA-GFDL/FMS/blob/2ff5e09985ffa7a5a8a51c40c4ffc5619128bf8e/mpp/include/mpp_io_read.inc#L874-L875

The logic is **never** going to be true. 
`bounds_name` is read as the `bounds` attribute to the variable (`lat`) (which has the null character) and `name` is obtained as the name of the variable in the file. 

so you will end up with something like this and the *bnds will never be read
```f90
if (trim('lat_bnds') .eq. trim('lat_bnds\000'))
```

Then when you call `get_axis_bounds` in [data_override](https://github.com/NOAA-GFDL/FMS/blob/2ff5e09985ffa7a5a8a51c40c4ffc5619128bf8e/data_override/data_override.F90#L2305), `bounds_found` will never be true so you will end up in this if clause, which calculates the bnds instead of reading them. 

https://github.com/NOAA-GFDL/FMS/blob/2ff5e09985ffa7a5a8a51c40c4ffc5619128bf8e/axis_utils/axis_utils.F90#L176-L191

@nikizadehgfdl fms2io actually reads the bnds so that explains the "answer changes" in the "OM4p5_BLING_JRA55do1.4_cycle1" experiment. The mpp_io answers are "wrong" 

I printed the `lat_bnds` and this is what I get from [mpp_io](https://github.com/NOAA-GFDL/FMS/blob/2ff5e09985ffa7a5a8a51c40c4ffc5619128bf8e/data_override/data_override.F90#L2312):
**-89.8485462603989**       -89.2916328408143
  -88.7330749838675       -88.1725011410287       -87.6114148125410    
  -87.0501193471326       -86.4887175679754       -85.9272542346880  
  -85.3657520244323       -84.8042236749131       -84.2426768993163
  -83.6811166442275       -83.1195462292232       -82.5579679655716
  -81.9963835123501       -81.434794091489**6** ... 

and this if from [fms2io](https://github.com/NOAA-GFDL/FMS/blob/2ff5e09985ffa7a5a8a51c40c4ffc5619128bf8e/data_override/data_override.F90#L825):
**-90.0000000000000**       -89.2916328408143     
  -88.7330749838675       -88.1725011410287       -87.6114148125410     
  -87.0501193471326       -86.4887175679754       -85.9272542346880     
  -85.3657520244323       -84.8042236749131       -84.2426768993163     
  -83.6811166442275       -83.1195462292232       -82.5579679655716     
  -81.9963835123501       -81.434794091489**7** ...

The main and probably the most significant difference is the first point, but there are also some places where the last decimal place is different. 

**To Reproduce**
Try to read *_bnds with any file that has the c null character in the *_bounds attribute and you will see it is not exactly the same as in the file. 

**Expected behavior**
If the *_bnds exist in the file then mpp_io should read it from the file, otherwise do the calculation in [get_axis_bounds](https://github.com/NOAA-GFDL/FMS/blob/2ff5e09985ffa7a5a8a51c40c4ffc5619128bf8e/axis_utils/axis_utils.F90#L176)

**System Environment**
Describe the system environment, include:
This happens in any system. 

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by uramirez8707**
**Description**
This PR adds a namelist flag to data_override, "reproduce_null_char_bug". If it is set to true it will reproduce the mpp_io_bug where the *_bnds were calculated instead of read in. 

This should reproduce mpp_io answers for experiments calling data_override on netcdf files with a null character in the string attributes. 

Fixes #578 

**How Has This Been Tested?**
Passes `make check` in skylake and travis. 
Ran a small separate unit test. 

@nikizadehgfdl Please confirm that this works and reproduces mpp_io answers. You need to add `reproduce_null_char_bug = .true.` to your data_override_nml.

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Review by bensonr — APPROVED** [review]

File: axis_utils/axis_utils2.F90 [hunk 1]
```diff
@@ -144,11 +144,12 @@ subroutine get_axis_cart(fileobj, axisname, cart)
     end if
   end subroutine get_axis_cart
 
-  subroutine axis_edges(fileobj, name, edge_data)
+  subroutine axis_edges(fileobj, name, edge_data, reproduce_null_char_bug_flag)
 
   class(FmsNetcdfFile_t), intent(in) :: fileobj
   character(len=*), intent(in) :: name
   class(*), dimension(:), intent(out) :: edge_data
+  logical, intent(in), optional :: reproduce_null_char_bug_flag
```
Comment (thomas-robinson) [inline axis_utils/axis_utils2.F90:152]: Please document the variable here

File: axis_utils/axis_utils2.F90 [hunk 2]
```diff
@@ -159,6 +160,7 @@ subroutine axis_edges(fileobj, name, edge_data)
   real(kind=real64), dimension(:,:), allocatable :: r642d
   integer :: i
   integer :: n
+  logical :: reproduce_null_char_bug
```
Comment (thomas-robinson) [inline axis_utils/axis_utils2.F90:163]: Please document the variable here

File: fms2_io/include/get_variable_attribute.inc [hunk 3]
```diff
@@ -31,18 +31,28 @@ subroutine get_variable_attribute_0d(fileobj, variable_name, attribute_name, &
                                              !! "I/O root" ranks.
                                              !! The broadcast will be done
                                              !! by default.
+  logical, intent(in), optional :: reproduce_null_char_bug_flag
```
Comment (thomas-robinson) [inline fms2_io/include/get_variable_attribute.inc:34]: Document the variable here in doxygen style

File: fms2_io/include/get_variable_attribute.inc [hunk 4]
```diff
@@ -31,18 +31,28 @@ subroutine get_variable_attribute_0d(fileobj, variable_name, attribute_name, &
...
 
   integer :: varid
   integer :: err
   integer :: j
   character(len=1024), dimension(1) :: charbuf !< 1D Character buffer
+  logical :: reproduce_null_char_bug
```
Comment (thomas-robinson) [inline fms2_io/include/get_variable_attribute.inc:40]: Please document this variable in doxygen style

File: fms2_io/include/get_variable_attribute.inc [hunk 5]
```diff
@@ -64,7 +74,11 @@ subroutine get_variable_attribute_0d(fileobj, variable_name, attribute_name, &
   select type(attribute_value)
     type is (character(len=*))
       call mpp_broadcast(charbuf, len(charbuf), fileobj%io_root, pelist=fileobj%pelist)
-      call string_copy(attribute_value, charbuf(1), check_for_null=.true.)
+      if (reproduce_null_char_bug) then
+         call string_copy(attribute_value, charbuf(1), check_for_null=.false.)
+      else
+         call string_copy(attribute_value, charbuf(1), check_for_null=.true.)
+      endif
```
Comment (thomas-robinson) [inline fms2_io/include/get_variable_attribute.inc:81]: Is this for all attributes?  I thought the only ones that mattered were `lat_bnds` and `lon_bnds` for this bug.

**Review by thomas-robinson — CHANGES_REQUESTED** [review]
See https://github.com/NOAA-GFDL/FMS/blob/master/CODE_STYLE.md#functions
- Inline doxygen descriptions for all arguments, except the result variable.

Please document subroutine arguments.

File: fms2_io/include/get_variable_attribute.inc (hunk 5 above)
Comment (uramirez8707) [reply fms2_io/include/get_variable_attribute.inc:81]: `reproduce_null_char_bug` is only going to be true if it is passed in the `get_variable_attribute`. 

This only happens in axis_utils:
https://github.com/NOAA-GFDL/FMS/blob/fa87962ce034cb2cf985e9cfe19d8983ba97273f/axis_utils/axis_utils2.F90#L180
https://github.com/NOAA-GFDL/FMS/blob/fa87962ce034cb2cf985e9cfe19d8983ba97273f/axis_utils/axis_utils2.F90#L190

So it is not for all attributes...

**Review by wrongkindofdoctor — DISMISSED** [review]

**Comment by nikizadehgfdl** [comment]
Thanks @uramirez8707 , this worked as advertised, reproducing old answers for a JRA run.

**Review by thomas-robinson — APPROVED** [review]

**Review by bensonr — APPROVED** [review]

**Review by wrongkindofdoctor — APPROVED** [review]

**Pull request merged**